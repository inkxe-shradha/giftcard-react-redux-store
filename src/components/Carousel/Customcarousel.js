import "react-responsive-carousel/lib/styles/carousel.min.css"; // requires a loader
import { Carousel } from "react-responsive-carousel";
import "./index.scss";
import { useNavigate } from "react-router-dom";
const CustomCarousel = () => {
  const settings = {
    autoPlay: true,
    dynamicHeight: false,
    showThumbs: false,
  };
  const navigate = useNavigate();
  return (
    <div className="container-fluid">
      <Carousel {...settings}>
        <div>
          <img
            className="carousel-img"
            src="https://source.unsplash.com/featured/900x450?amazon"
            alt="amazon"
          />
          <button onClick={() => navigate("/gift-cards")} className="legend">
            {" "}
            Grab it
          </button>
        </div>
        <div>
          <img
            className="carousel-img"
            src="https://source.unsplash.com/featured/900x450?coupon"
            alt="flipkart"
          />
          <button onClick={() => navigate("/gift-cards")} className="legend">
            {" "}
            Grab it
          </button>
        </div>
        <div>
          <img
            className="carousel-img"
            src="https://source.unsplash.com/featured/900x450?google"
            alt="google"
          />
          <button onClick={() => navigate("/gift-cards")} className="legend">
            {" "}
            Grab it
          </button>
        </div>
      </Carousel>
    </div>
  );
};

export default CustomCarousel;
