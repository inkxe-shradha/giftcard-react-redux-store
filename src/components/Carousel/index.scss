.carousel-img {
    width: 100%;
    height: 100%;
    max-height: 450px;
    background-size: cover;
    background-position: center;
    background-repeat: no-repeat;
}

.legend {
    cursor: pointer;
}