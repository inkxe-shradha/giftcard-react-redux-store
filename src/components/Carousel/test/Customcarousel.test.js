import { render } from "@testing-library/react";
import Carousel from "../Customcarousel";
import { BrowserRouter } from "react-router-dom";
import { mount } from "enzyme";

const mockedUsedNavigate = jest.fn();

// Mocks like this need to be configured at the top level

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockedUsedNavigate,
}));

describe("Carousel testing suite cases", () => {
  it("Should able to capture the snapshot of the Carousel", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Carousel />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("Should have atleast 3 carousel images", () => {
    const view = mount(
      <BrowserRouter>
        <Carousel />
      </BrowserRouter>
    );
    expect(view.find(".carousel-img").length).toBeGreaterThanOrEqual(3);
  });

  it("Should navigate to the gift-card page if view more button clicked", () => {
    const view = mount(
      <BrowserRouter>
        <Carousel />
      </BrowserRouter>
    );
    view.find(".legend").at(0).simulate("click");
    expect(mockedUsedNavigate).toHaveBeenCalledWith("/gift-cards");
  });
});
