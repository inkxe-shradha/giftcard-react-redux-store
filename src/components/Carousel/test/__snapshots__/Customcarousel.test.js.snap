// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel testing suite cases Should able to capture the snapshot of the Carousel 1`] = `
<DocumentFragment>
  <div
    class="container-fluid"
  >
    <div
      class="carousel-root"
    >
      <div
        class="carousel carousel-slider"
        style="width: 100%;"
      >
        <ul
          class="control-dots"
        >
          <li
            aria-label="slide item 1"
            class="dot selected"
            role="button"
            tabindex="0"
            value="0"
          />
          <li
            aria-label="slide item 2"
            class="dot"
            role="button"
            tabindex="0"
            value="1"
          />
          <li
            aria-label="slide item 3"
            class="dot"
            role="button"
            tabindex="0"
            value="2"
          />
        </ul>
        <button
          aria-label="previous slide / item"
          class="control-arrow control-prev control-disabled"
          type="button"
        />
        <div
          class="slider-wrapper axis-horizontal"
        >
          <ul
            class="slider animated"
            style="transform: translate3d(0,0,0); transition-duration: 350ms;"
          >
            <li
              class="slide selected previous"
            >
              <div>
                <img
                  alt="amazon"
                  class="carousel-img"
                  src="https://source.unsplash.com/featured/900x450?amazon"
                />
                <button
                  class="legend"
                >
                   Grab it
                </button>
              </div>
            </li>
            <li
              class="slide"
            >
              <div>
                <img
                  alt="flipkart"
                  class="carousel-img"
                  src="https://source.unsplash.com/featured/900x450?coupon"
                />
                <button
                  class="legend"
                >
                   Grab it
                </button>
              </div>
            </li>
            <li
              class="slide"
            >
              <div>
                <img
                  alt="google"
                  class="carousel-img"
                  src="https://source.unsplash.com/featured/900x450?google"
                />
                <button
                  class="legend"
                >
                   Grab it
                </button>
              </div>
            </li>
          </ul>
        </div>
        <button
          aria-label="next slide / item"
          class="control-arrow control-next"
          type="button"
        />
        <p
          class="carousel-status"
        >
          1 of 3
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
